libsolidity/ast/AsmJsonImporter.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author julius <djudju@protonmail.com>
+ * @date 2019
+ * Converts an inlineAssembly AST from JSON format to AsmData
+
+ */
+
+#include <libsolidity/ast/AsmJsonImporter.h>
+#include <libsolidity/ast/ASTJsonImporter.h>
+#include <libsolidity/ast/Types.h>
+#include <libyul/AsmData.h>
+#include <libyul/AsmDataForward.h>
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/Scanner.h>
+#include <vector>
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+
+using SourceLocation = langutil::SourceLocation;
+
+SourceLocation const AsmJsonImporter::createSourceLocation(Json::Value const& _node)
```

Comment:
There is still no clarity in this function. Can you just create a helper in `SourceLocation` for parsing it?

Reply:
made it more clear and added a comment about the expected input

Reply:
It turns out you did that for `ASTJsonImporter::createSourceLocation` and I was confused because this place here never changed. The two functions seems to be duplicated but look very different now. This is even more of a reason to have a helper. Can you helpers for both parsing & serialisation and create a new PR with a uni test?

Reply:
Still a blocker, i would say.

Reply:
completely missed this